cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginConfig.java

```diff
@@ -37,4 +40,23 @@
   public final PluginProperties getProperties() {
     return properties;
   }
+
+  /**
+   * Returns true if property value is a macro; false otherwise.
+   * @param fieldName
+   * @return
+   */
+  public boolean isMacro(String fieldName) {
+    try {
+      Annotation annotation = this.getClass().getDeclaredField(fieldName).getAnnotation(Macro.class);
```

Comment:
this can throw a NPE if the field passed in is not a declared field.  I think it would be better if all the set of fields with macros are evaluated beforehand, then this method just checks if the given field name is contained in the set. Then you wouldn't need to expose MacroUtils in api either.